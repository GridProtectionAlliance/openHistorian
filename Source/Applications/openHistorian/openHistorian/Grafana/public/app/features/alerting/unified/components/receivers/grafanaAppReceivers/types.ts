import { Receiver } from '../../../../../../plugins/datasource/alertmanager/types';
import { SupportedPlugin } from '../../../types/pluginBridges';

export interface AmRouteReceiver {
  label: string;
  value: string;
  grafanaAppReceiverType?: SupportedPlugin;
}

export interface ReceiverWithTypes extends Receiver {
  grafanaAppReceiverType?: SupportedPlugin;
}
export const GRAFANA_APP_RECEIVERS_SOURCE_IMAGE: Record<SupportedPlugin, string> = {
  [SupportedPlugin.OnCall]: 'public/img/alerting/oncall_logo.svg',

  [SupportedPlugin.Incident]: '',
  [SupportedPlugin.MachineLearning]: '',
};
