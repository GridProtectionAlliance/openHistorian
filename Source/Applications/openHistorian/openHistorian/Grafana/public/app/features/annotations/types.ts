import { AnnotationEvent, PanelData, TimeRange } from '@grafana/data';

import { DashboardModel, PanelModel } from '../dashboard/state';

export interface AnnotationQueryOptions {
  dashboard: DashboardModel;
  panel: PanelModel;
  range: TimeRange;
}

export interface AnnotationQueryResponse {
  /**
   * The processed annotation events
   */
  events?: AnnotationEvent[];

  /**
   * The original panel response
   */
  panelData?: PanelData;
}

export interface AnnotationTag {
  /**
   * The tag name
   */
  tag: string;
  /**
   * The number of occurrences of that tag
   */
  count: number;
}

export interface AnnotationTagsResponse {
  result: {
    tags: AnnotationTag[];
  };
}
