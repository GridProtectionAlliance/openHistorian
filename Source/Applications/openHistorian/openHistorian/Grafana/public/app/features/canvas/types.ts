import { ColorDimensionConfig, ResourceDimensionConfig } from 'app/features/dimensions/types';

export interface Placement {
  top?: number;
  left?: number;
  right?: number;
  bottom?: number;

  width?: number;
  height?: number;
}

export interface Constraint {
  horizontal?: HorizontalConstraint;
  vertical?: VerticalConstraint;
}

export enum HorizontalConstraint {
  Left = 'left',
  Right = 'right',
  LeftRight = 'leftright',
  Center = 'center',
  Scale = 'scale',
}

export enum VerticalConstraint {
  Top = 'top',
  Bottom = 'bottom',
  TopBottom = 'topbottom',
  Center = 'center',
  Scale = 'scale',
}

export enum BackgroundImageSize {
  Original = 'original',
  Contain = 'contain',
  Cover = 'cover',
  Fill = 'fill',
  Tile = 'tile',
}

export interface BackgroundConfig {
  color?: ColorDimensionConfig;
  image?: ResourceDimensionConfig;
  size?: BackgroundImageSize;
}

export interface LineConfig {
  color?: ColorDimensionConfig;
  width?: number;
}

export enum QuickPlacement {
  Top = 'top',
  Bottom = 'bottom',
  Left = 'left',
  Right = 'right',
  HorizontalCenter = 'hcenter',
  VerticalCenter = 'vcenter',
}
