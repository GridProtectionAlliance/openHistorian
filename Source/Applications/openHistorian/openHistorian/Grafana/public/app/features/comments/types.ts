export interface MessagePacket {
  event: string;
  commentCreated: Message;
}

export interface Message {
  id: number;
  content: string;
  created: number;
  userId: number;
  user: User;
}

// TODO: Interface may exist elsewhere
export interface User {
  id: number;
  name: string;
  login: string;
  email: string;
  avatarUrl: string;
}
