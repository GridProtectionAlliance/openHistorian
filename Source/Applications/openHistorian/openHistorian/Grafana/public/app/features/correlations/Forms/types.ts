import { CorrelationConfig } from '../types';

export interface FormDTO {
  sourceUID: string;
  targetUID: string;
  label: string;
  description: string;
  config: CorrelationConfig;
}

export type EditFormDTO = Omit<FormDTO, 'targetUID' | 'sourceUID'>;
