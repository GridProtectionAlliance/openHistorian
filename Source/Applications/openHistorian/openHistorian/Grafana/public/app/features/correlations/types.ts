import { DataLinkTransformationConfig } from '@grafana/data';

export interface AddCorrelationResponse {
  correlation: Correlation;
}

export type GetCorrelationsResponse = Correlation[];

type CorrelationConfigType = 'query';

export interface CorrelationConfig {
  field: string;
  target: object;
  type: CorrelationConfigType;
  transformations?: DataLinkTransformationConfig[];
}

export interface Correlation {
  uid: string;
  sourceUID: string;
  targetUID: string;
  label?: string;
  description?: string;
  config: CorrelationConfig;
}

export type RemoveCorrelationParams = Pick<Correlation, 'sourceUID' | 'uid'>;
export type CreateCorrelationParams = Omit<Correlation, 'uid'>;
export type UpdateCorrelationParams = Omit<Correlation, 'targetUID'>;
