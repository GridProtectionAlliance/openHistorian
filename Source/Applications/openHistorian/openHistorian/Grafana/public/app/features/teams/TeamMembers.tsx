import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { SlideDown } from 'app/core/components/Animations/SlideDown';
import { UserPicker } from 'app/core/components/Select/UserPicker';
import { TagBadge } from 'app/core/components/TagFilter/TagBadge';
import { TeamMember, User } from 'app/types';
import { addTeamMember } from './state/actions';
import { getSearchMemberQuery, isSignedInUserTeamAdmin } from './state/selectors';
import { FilterInput } from 'app/core/components/FilterInput/FilterInput';
import { WithFeatureToggle } from 'app/core/components/WithFeatureToggle';
import { config } from 'app/core/config';
import { contextSrv, User as SignedInUser } from 'app/core/services/context_srv';
import TeamMemberRow from './TeamMemberRow';
import { setSearchMemberQuery } from './state/reducers';

export interface Props {
  members: TeamMember[];
  searchMemberQuery: string;
  addTeamMember: typeof addTeamMember;
  setSearchMemberQuery: typeof setSearchMemberQuery;
  syncEnabled: boolean;
  editorsCanAdmin?: boolean;
  signedInUser?: SignedInUser;
}

export interface State {
  isAdding: boolean;
  newTeamMember?: User;
}

export class TeamMembers extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { isAdding: false, newTeamMember: null };
  }

  onSearchQueryChange = (value: string) => {
    this.props.setSearchMemberQuery(value);
  };

  onToggleAdding = () => {
    this.setState({ isAdding: !this.state.isAdding });
  };

  onUserSelected = (user: User) => {
    this.setState({ newTeamMember: user });
  };

  onAddUserToTeam = async () => {
    this.props.addTeamMember(this.state.newTeamMember.id);
    this.setState({ newTeamMember: null });
  };

  renderLabels(labels: string[]) {
    if (!labels) {
      return <td />;
    }

    return (
      <td>
        {labels.map(label => (
          <TagBadge key={label} label={label} removeIcon={false} count={0} onClick={() => {}} />
        ))}
      </td>
    );
  }

  render() {
    const { isAdding } = this.state;
    const { searchMemberQuery, members, syncEnabled, editorsCanAdmin, signedInUser } = this.props;
    const isTeamAdmin = isSignedInUserTeamAdmin({ members, editorsCanAdmin, signedInUser });

    return (
      <div>
        <div className="page-action-bar">
          <div className="gf-form gf-form--grow">
            <FilterInput
              labelClassName="gf-form--has-input-icon gf-form--grow"
              inputClassName="gf-form-input"
              placeholder="Search members"
              value={searchMemberQuery}
              onChange={this.onSearchQueryChange}
            />
          </div>

          <div className="page-action-bar__spacer" />

          <button
            className="btn btn-primary pull-right"
            onClick={this.onToggleAdding}
            disabled={isAdding || !isTeamAdmin}
          >
            Add member
          </button>
        </div>

        <SlideDown in={isAdding}>
          <div className="cta-form">
            <button className="cta-form__close btn btn-transparent" onClick={this.onToggleAdding}>
              <i className="fa fa-close" />
            </button>
            <h5>Add team member</h5>
            <div className="gf-form-inline">
              <UserPicker onSelected={this.onUserSelected} className="min-width-30" />
              {this.state.newTeamMember && (
                <button className="btn btn-primary gf-form-btn" type="submit" onClick={this.onAddUserToTeam}>
                  Add to team
                </button>
              )}
            </div>
          </div>
        </SlideDown>

        <div className="admin-list-table">
          <table className="filter-table filter-table--hover form-inline">
            <thead>
              <tr>
                <th />
                <th>Login</th>
                <th>Email</th>
                <th>Name</th>
                <WithFeatureToggle featureToggle={editorsCanAdmin}>
                  <th>Permission</th>
                </WithFeatureToggle>
                {syncEnabled && <th />}
                <th style={{ width: '1%' }} />
              </tr>
            </thead>
            <tbody>
              {members &&
                members.map(member => (
                  <TeamMemberRow
                    key={member.userId}
                    member={member}
                    syncEnabled={syncEnabled}
                    editorsCanAdmin={editorsCanAdmin}
                    signedInUserIsTeamAdmin={isTeamAdmin}
                  />
                ))}
            </tbody>
          </table>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state: any) {
  return {
    searchMemberQuery: getSearchMemberQuery(state.team),
    editorsCanAdmin: config.editorsCanAdmin, // this makes the feature toggle mockable/controllable from tests,
    signedInUser: contextSrv.user, // this makes the feature toggle mockable/controllable from tests,
  };
}

const mapDispatchToProps = {
  addTeamMember,
  setSearchMemberQuery,
};

export default connect(mapStateToProps, mapDispatchToProps)(TeamMembers);
