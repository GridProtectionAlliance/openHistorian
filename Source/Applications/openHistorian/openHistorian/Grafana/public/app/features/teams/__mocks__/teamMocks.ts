import { Team, TeamGroup, TeamMember, TeamPermissionLevel } from 'app/types';

export const getMultipleMockTeams = (numberOfTeams: number): Team[] => {
  const teams: Team[] = [];
  for (let i = 1; i <= numberOfTeams; i++) {
    teams.push({
      id: i,
      name: `test-${i}`,
      avatarUrl: 'some/url/',
      email: `test-${i}@test.com`,
      memberCount: i,
      permission: TeamPermissionLevel.Member,
    });
  }

  return teams;
};

export const getMockTeam = (): Team => {
  return {
    id: 1,
    name: 'test',
    avatarUrl: 'some/url/',
    email: 'test@test.com',
    memberCount: 1,
    permission: TeamPermissionLevel.Member,
  };
};

export const getMockTeamMembers = (amount: number, teamAdminId: number): TeamMember[] => {
  const teamMembers: TeamMember[] = [];

  for (let i = 1; i <= amount; i++) {
    teamMembers.push({
      userId: i,
      teamId: 1,
      avatarUrl: 'some/url/',
      email: 'test@test.com',
      name: 'testName',
      login: `testUser-${i}`,
      labels: ['label 1', 'label 2'],
      permission: i === teamAdminId ? TeamPermissionLevel.Admin : TeamPermissionLevel.Member,
    });
  }

  return teamMembers;
};

export const getMockTeamMember = (): TeamMember => {
  return {
    userId: 1,
    teamId: 1,
    avatarUrl: 'some/url/',
    email: 'test@test.com',
    name: 'testName',
    login: 'testUser',
    labels: [],
    permission: TeamPermissionLevel.Member,
  };
};

export const getMockTeamGroups = (amount: number): TeamGroup[] => {
  const groups: TeamGroup[] = [];

  for (let i = 1; i <= amount; i++) {
    groups.push({
      groupId: `group-${i}`,
      teamId: 1,
    });
  }

  return groups;
};
