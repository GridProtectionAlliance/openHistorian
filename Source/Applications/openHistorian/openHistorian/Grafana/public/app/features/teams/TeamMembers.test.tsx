import React from 'react';
import { shallow } from 'enzyme';
import { TeamMembers, Props, State } from './TeamMembers';
import { TeamMember, OrgRole } from '../../types';
import { getMockTeamMembers } from './__mocks__/teamMocks';
import { User } from 'app/core/services/context_srv';

const signedInUserId = 1;

const setup = (propOverrides?: object) => {
  const props: Props = {
    members: [] as TeamMember[],
    searchMemberQuery: '',
    setSearchMemberQuery: jest.fn(),
    addTeamMember: jest.fn(),
    syncEnabled: false,
    editorsCanAdmin: false,
    signedInUser: {
      id: signedInUserId,
      isGrafanaAdmin: false,
      orgRole: OrgRole.Viewer,
    } as User,
  };

  Object.assign(props, propOverrides);

  const wrapper = shallow(<TeamMembers {...props} />);
  const instance = wrapper.instance() as TeamMembers;

  return {
    wrapper,
    instance,
  };
};

describe('Render', () => {
  it('should render component', () => {
    const { wrapper } = setup({});

    expect(wrapper).toMatchSnapshot();
  });

  it('should render team members', () => {
    const { wrapper } = setup({ members: getMockTeamMembers(5, 5) });

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Functions', () => {
  describe('on search member query change', () => {
    it('it should call setSearchMemberQuery', () => {
      const { instance } = setup({});

      instance.onSearchQueryChange('member');

      expect(instance.props.setSearchMemberQuery).toHaveBeenCalledWith('member');
    });
  });

  describe('on add user to team', () => {
    const { wrapper, instance } = setup({});
    const state = wrapper.state() as State;

    state.newTeamMember = {
      id: 1,
      label: '',
      avatarUrl: '',
      login: '',
      name: '',
      email: '',
    };

    instance.onAddUserToTeam();

    expect(instance.props.addTeamMember).toHaveBeenCalledWith(1);
  });
});
