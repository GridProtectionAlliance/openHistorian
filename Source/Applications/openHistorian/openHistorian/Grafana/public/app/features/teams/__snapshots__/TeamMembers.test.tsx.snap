// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div>
  <div
    className="page-action-bar"
  >
    <div
      className="gf-form gf-form--grow"
    >
      <Component
        inputClassName="gf-form-input"
        labelClassName="gf-form--has-input-icon gf-form--grow"
        onChange={[Function]}
        placeholder="Search members"
        value=""
      />
    </div>
    <div
      className="page-action-bar__spacer"
    />
    <button
      className="btn btn-primary pull-right"
      disabled={false}
      onClick={[Function]}
    >
      Add member
    </button>
  </div>
  <Component
    in={false}
  >
    <div
      className="cta-form"
    >
      <button
        className="cta-form__close btn btn-transparent"
        onClick={[Function]}
      >
        <Icon
          name="times"
        />
      </button>
      <h5>
        Add team member
      </h5>
      <div
        className="gf-form-inline"
      >
        <UserPicker
          className="min-width-30"
          onSelected={[Function]}
        />
      </div>
    </div>
  </Component>
  <div
    className="admin-list-table"
  >
    <table
      className="filter-table filter-table--hover form-inline"
    >
      <thead>
        <tr>
          <th />
          <th>
            Login
          </th>
          <th>
            Email
          </th>
          <th>
            Name
          </th>
          <Component
            featureToggle={false}
          >
            <th>
              Permission
            </th>
          </Component>
          <th
            style={
              Object {
                "width": "1%",
              }
            }
          />
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`Render should render team members 1`] = `
<div>
  <div
    className="page-action-bar"
  >
    <div
      className="gf-form gf-form--grow"
    >
      <Component
        inputClassName="gf-form-input"
        labelClassName="gf-form--has-input-icon gf-form--grow"
        onChange={[Function]}
        placeholder="Search members"
        value=""
      />
    </div>
    <div
      className="page-action-bar__spacer"
    />
    <button
      className="btn btn-primary pull-right"
      disabled={false}
      onClick={[Function]}
    >
      Add member
    </button>
  </div>
  <Component
    in={false}
  >
    <div
      className="cta-form"
    >
      <button
        className="cta-form__close btn btn-transparent"
        onClick={[Function]}
      >
        <Icon
          name="times"
        />
      </button>
      <h5>
        Add team member
      </h5>
      <div
        className="gf-form-inline"
      >
        <UserPicker
          className="min-width-30"
          onSelected={[Function]}
        />
      </div>
    </div>
  </Component>
  <div
    className="admin-list-table"
  >
    <table
      className="filter-table filter-table--hover form-inline"
    >
      <thead>
        <tr>
          <th />
          <th>
            Login
          </th>
          <th>
            Email
          </th>
          <th>
            Name
          </th>
          <Component
            featureToggle={false}
          >
            <th>
              Permission
            </th>
          </Component>
          <th
            style={
              Object {
                "width": "1%",
              }
            }
          />
        </tr>
      </thead>
      <tbody>
        <Connect(TeamMemberRow)
          editorsCanAdmin={false}
          key="1"
          member={
            Object {
              "avatarUrl": "some/url/",
              "email": "test@test.com",
              "labels": Array [
                "label 1",
                "label 2",
              ],
              "login": "testUser-1",
              "name": "testName",
              "permission": 0,
              "teamId": 1,
              "userId": 1,
            }
          }
          signedInUserIsTeamAdmin={true}
          syncEnabled={false}
        />
        <Connect(TeamMemberRow)
          editorsCanAdmin={false}
          key="2"
          member={
            Object {
              "avatarUrl": "some/url/",
              "email": "test@test.com",
              "labels": Array [
                "label 1",
                "label 2",
              ],
              "login": "testUser-2",
              "name": "testName",
              "permission": 0,
              "teamId": 1,
              "userId": 2,
            }
          }
          signedInUserIsTeamAdmin={true}
          syncEnabled={false}
        />
        <Connect(TeamMemberRow)
          editorsCanAdmin={false}
          key="3"
          member={
            Object {
              "avatarUrl": "some/url/",
              "email": "test@test.com",
              "labels": Array [
                "label 1",
                "label 2",
              ],
              "login": "testUser-3",
              "name": "testName",
              "permission": 0,
              "teamId": 1,
              "userId": 3,
            }
          }
          signedInUserIsTeamAdmin={true}
          syncEnabled={false}
        />
        <Connect(TeamMemberRow)
          editorsCanAdmin={false}
          key="4"
          member={
            Object {
              "avatarUrl": "some/url/",
              "email": "test@test.com",
              "labels": Array [
                "label 1",
                "label 2",
              ],
              "login": "testUser-4",
              "name": "testName",
              "permission": 0,
              "teamId": 1,
              "userId": 4,
            }
          }
          signedInUserIsTeamAdmin={true}
          syncEnabled={false}
        />
        <Connect(TeamMemberRow)
          editorsCanAdmin={false}
          key="5"
          member={
            Object {
              "avatarUrl": "some/url/",
              "email": "test@test.com",
              "labels": Array [
                "label 1",
                "label 2",
              ],
              "login": "testUser-5",
              "name": "testName",
              "permission": 4,
              "teamId": 1,
              "userId": 5,
            }
          }
          signedInUserIsTeamAdmin={true}
          syncEnabled={false}
        />
      </tbody>
    </table>
  </div>
</div>
`;
